test/e2e/framework/pod/utils.go

```diff
@@ -164,11 +184,15 @@ func MixinRestrictedPodSecurity(pod *v1.Pod) error {
 			pod.Spec.SecurityContext.RunAsNonRoot = pointer.BoolPtr(true)
 		}
 		if pod.Spec.SecurityContext.RunAsUser == nil {
-			pod.Spec.SecurityContext.RunAsUser = pointer.Int64Ptr(DefaultNonRootUser)
+			pod.Spec.SecurityContext.RunAsUser = GetDefaultNonRootUser()
 		}
 		if pod.Spec.SecurityContext.SeccompProfile == nil {
 			pod.Spec.SecurityContext.SeccompProfile = &v1.SeccompProfile{Type: v1.SeccompProfileTypeRuntimeDefault}
 		}
+		if NodeOSDistroIs("windows") && pod.Spec.SecurityContext.WindowsOptions == nil {
+			pod.Spec.SecurityContext.WindowsOptions = &v1.WindowsSecurityContextOptions{}
+			pod.Spec.SecurityContext.WindowsOptions.RunAsUserName = pointer.StringPtr(DefaultNonRootUserName)
+		}
 	}
 	for i := range pod.Spec.Containers {
 		mixinRestrictedContainerSecurityContext(&pod.Spec.Containers[i])
```

Comment:
Do you want to add similar securityContext checks for containers? It may not be a problem unless pod's explicitly have containers that have Windows options set.

Reply:
This doesn't look to set the users for the containers currently, so left it out for now.